hrms/hr/doctype/hr_settings/hr_settings.json

```diff
@@ -329,13 +330,19 @@
    "fieldname": "unlink_payment_on_cancellation_of_employee_advance",
    "fieldtype": "Check",
    "label": " Unlink Payment on Cancellation of Employee Advance"
+  },
+  {
+   "default": "1",
+   "fieldname": "allow_self_leave_approval",
+   "fieldtype": "Check",
+   "label": "Enable Self-Approval For Leaves"
```

Comment:
```suggestion
   "label": "Allow self approval for Leave Application"
```

Reply:
Based on this setting, Approval field should also become read-only - client-side change